test/integration/multicluster/multicluster-traffic/mc_traffic_test.go

```diff
@@ -0,0 +1,369 @@
+		// the gateway keeps track of the mirror services.
+		tgtWorkloadRollouts := map[string]testutil.DeploySpec{
+			"nginx-deploy": {Namespace: "linkerd-nginx-gateway-deploy", Replicas: 1},
+		}
+		TestHelper.WaitRolloutWithContext(t, tgtWorkloadRollouts, contexts["tgt"])
+	})
+
+	timeout := time.Minute
+	err := TestHelper.RetryFor(timeout, func() error {
+		out, err := TestHelper.LinkerdRun("--context="+contexts["src"], "multicluster", "gateways")
+		if err != nil {
+			return err
+		}
+		rows := strings.Split(out, "\n")
+		if len(rows) < 2 {
+			return errors.New("response is empty")
+		}
+		fields := strings.Fields(rows[1])
+		if len(fields) < 6 {
+			return fmt.Errorf("unexpected number of columns: %d", len(fields))
+		}
+		if fields[0] != "target" {
+			return fmt.Errorf("unexpected target cluster name: %s", fields[0])
+		}
+		if fields[1] != "True" {
+			return errors.New("target cluster is not alive")
+		}
+		if fields[2] != "1" {
+			return fmt.Errorf("invalid NUM_SVC: %s", fields[2])
+		}
+
+		return nil
+	})
+	if err != nil {
+		testutil.AnnotatedFatal(t, fmt.Sprintf("'linkerd multicluster gateways' command timed-out (%s)", timeout), err)
+	}
+}
+
+// TestCheckAfterRepairEndpoints calls `linkerd mc check` again after 1 minute,
+// so that the RepairEndpoints event has already been processed, making sure
+// that resyncing didn't break things.
+// CheckGateway will make use of nginx-gateway-deploy to see whether check
+// result returns number of mirror services.
+func TestCheckGatewayAfterRepairEndpoints(t *testing.T) {
+	// Re-build the clientset with the source context
+	if err := TestHelper.SwitchContext(contexts["src"]); err != nil {
+		testutil.AnnotatedFatalf(t, "failed to rebuild helper clientset with new context", "failed to rebuild helper clientset with new context [%s]: %v", contexts["src"], err)
+	}
+	time.Sleep(time.Minute + 5*time.Second)
+	cmd := []string{"--context=" + contexts["src"], "multicluster", "check", "--wait=10s"}
+	timeout := 20 * time.Second
+	err := TestHelper.RetryFor(timeout, func() error {
+		out, err := TestHelper.LinkerdRun(cmd...)
+		if err != nil {
+			return fmt.Errorf("'linkerd multicluster check' command failed\n%s", out)
+		}
+
+		pods, err := TestHelper.KubernetesHelper.GetPods(context.Background(), "linkerd-multicluster", nil)
+		if err != nil {
+			testutil.AnnotatedFatal(t, fmt.Sprintf("failed to retrieve pods: %s", err), err)
+		}
+
+		tpl := template.Must(template.ParseFiles("testdata/check-gateway.multicluster.golden"))
+		versionErr := healthcheck.CheckProxyVersionsUpToDate(pods, version.Channels{})
+		versionErrMsg := ""
+		if versionErr != nil {
+			versionErrMsg = versionErr.Error()
+		}
+		vars := struct {
+			ProxyVersionErr string
+			HintURL         string
+			LinkName        string
+		}{
+			versionErrMsg,
+			healthcheck.HintBaseURL(TestHelper.GetVersion()),
+			"target", // name of linked cluster, not its context
+		}
```

Comment:
`linkerd multicluster check` already checks proxy versions. I assume this was already here, but I'm not sure why we have this explicit check here.

Reply:
Yeah, I think it's an unfortunate result of using `check` in tests. We always have to include the proxy versions and error message, more often than not the proxies will not run the latest version and the check will issue a warn. I've noticed we do this in a bunch of tests, all the more reasons to switch to `check -ojson`

Reply:
Yea so is this needed then? What I was implying is that `linkerd multicluster check` should already catch issues with versions.